### ISSUE TITLE
Unbound public symbol on ios target with 1.4.0 compose

### ISSUE THREAD

**Issue by afTrolle** [adrielcafe/voyager#132]
Hi, feel free to close this issue if it's not relevant.

I'm using voyager in a compose multiplatform project,
kotlin=1.8.20
compose=1.4.0

Everything is working fine for the android target, however there seems to be some issue with linkPodDebugFrameworkIosX64 task.

It seems related to this issue. 
https://github.com/JetBrains/compose-multiplatform/issues/2694

Which means that it will be resolved if or when voyager updates compose version.

Error-log: 
```
> Task :shared:linkPodDebugFrameworkIosX64 FAILED
w: Cannot infer a bundle ID from packages of source files and exported dependencies, use the bundle name instead: shared. Please specify the bundle ID explicitly using the -Xbinary=bundleId=<id> compiler flag.
e: There is still an unbound symbol after generation of IR module <shared>:
Cannot infer a bundle ID from packages of source files and exported dependencies, use the bundle name instead: shared. Please specify the bundle ID explicitly using the -Xbinary=bundleId=<id> compiler flag.

Unbound public symbol IrSimpleFunctionPublicSymbolImpl: cafe.adriel.voyager.core.screen/Screen.Content|-3823403984862487740[0]

This could happen if there are two libraries, where one library was compiled against the different version of the other library than the one currently used in the project. Please check that the project configuration is correct and has consistent versions of dependencies.

There is still an unbound symbol after generation of IR module <shared>:
Unbound public symbol IrSimpleFunctionPublicSymbolImpl: cafe.adriel.voyager.core.screen/Screen.Content|-3823403984862487740[0]

This could happen if there are two libraries, where one library was compiled against the different version of the other library than the one currently used in the project. Please check that the project configuration is correct and has consistent versions of dependencies.
```

**Comment by truongnguyenptit** [comment]
you must set internal like this
`internal fun MainComposeView(viewModel: ApplicationViewModel, modifier: Modifier = Modifier) `

**Comment by pablichjenkov** [comment]
In theory the internal issue got solved. However, there seems to be some cases still where exposing a Composable still causes the unbound symbol problem.
In my case, I was using a class property mark as Composable and got a similar message. I should file an issue.

**Comment by afTrolle** [comment]
I found two solutions to my project, 

- Forking project and bumping compose version solved it for me. 
- Implementations of Screen was propagated from the shared module - making them internal solved it.

I guess it might be because there are two versions of Composable annotations being exported from the shared module. 
One from voyager and one from directly importing  compose 1.4.0
I'm exporting a couple of Composables functions for previewing so there might be resolve issue there.

**Comment by DevSrSouza** [comment]
It should be bumped to 1.4.0 in the next version!

**Comment by afTrolle** [comment]
Thank you, @DevSrSouza great work with this library.

**Comment by jamesmbowler** [comment]
I'm still seeing this on latest version "1.0.0-rc06" when trying to run in ios simulator:

```error: There are still 3 unbound symbols after generation of IR module <shared>:
Unbound public symbol IrSimpleFunctionPublicSymbolImpl: cafe.adriel.voyager.navigator.tab/Tab.icon.<get-icon>|-5764684600227404327[0]
Unbound public symbol IrSimpleFunctionPublicSymbolImpl: cafe.adriel.voyager.navigator.tab/Tab.options.<get-options>|7766868889533212780[0]
Unbound public symbol IrSimpleFunctionPublicSymbolImpl: cafe.adriel.voyager.navigator.tab/Tab.title.<get-title>|371973571756195065[0]

**Comment by Syer10** [comment]
The tab navigator dependency hits a known Compose Compiler bug for iOS targets, we are waiting for Jetbrains to fix it, otherwise just remove the tab navigator dependency for now.

**Comment by jamesmbowler** [comment]
Thanks, I just created a version without online classes, works fine, thanks👍

**Comment by Kashif-E** [comment]
fixed in compose version 1.4.3 no need to use internal class/object any more
